waltz-test/src/main/python/waltz_ducktape/tests/validation/load_rebalancing_test.py

```diff
@@ -0,0 +1,97 @@
+        :param num_active_partitions: number of active partitions
+        :param delay_before_rebalancing: delay (in seconds) before rebalancing of nodes starts
+        :param expected_number_of_transactions: number of transactions expected to be stored on a waltz storage node
+        """
+
+        storage_admin_port = self.waltz_storage.admin_port
+        server_admin_port = self.waltz_server.port
+        storage_port = self.waltz_storage.port
+        storage_node = self.waltz_storage.nodes[randrange(len(self.waltz_storage.nodes))]
+        random_active_partition = randrange(num_active_partitions)
+
+        # Step 1: Start waltz cluster and execute validation command
+        self.verifiable_client.start(validation_cmd)
+
+        # Step 2: Wait till transactions get processed
+        wait_until(lambda: self.is_max_transaction_id_updated(self.get_host(storage_node.account.ssh_hostname, storage_admin_port),
+                                                              storage_port, random_active_partition, -1), timeout_sec=timeout)
+
+        # Step 3: Wait before staring torture test
+        sleep(delay_before_rebalancing)
+
+        # Step 4: Change assigned server node for random_active_partition
+        partition_current_server_index = self.get_server_node_idx(random_active_partition)
+
+        while True:
+            partition_future_server_index = randrange(len(self.waltz_server.nodes))
+            if not partition_current_server_index == partition_future_server_index:
+                break
+
+        server_node = self.waltz_server.nodes[partition_future_server_index]
+        server_node_hostname = server_node.account.ssh_hostname
+        server = self.get_host(server_node_hostname, server_admin_port)
+        self.server_cli.add_preferred_partition(server, random_active_partition)
+
+        assert self.get_server_node_idx(random_active_partition) == partition_future_server_index, \
+            "partition assignment for server nodes remained untouched"
+
+
+        # Step 5: Wait until verifiable client ends its task
+        wait_until(lambda: self.verifiable_client.task_complete() == True, timeout_sec=timeout,
+                   err_msg="verifiable_client failed to complete task in {} seconds.".format(timeout))
+
+
+        # Step 6: Assert all transactions are persistently stored. On each partition first transaction is indexed as 0
+        assert expected_number_of_transactions - num_active_partitions == self.sum_of_transactions(storage_node, storage_admin_port, storage_port, num_active_partitions), \
+            "number of transactions stored in storage partition does not match with all the transactions sent by producers. " \
+            "Client {}, Storage = {}" \
+                .format(expected_number_of_transactions, self.sum_of_transactions(node, storage_admin_port, port, num_active_partitions))
+
+
+    def sum_of_transactions(self, node, admin_port, port, num_active_partitions):
```

Comment:
The same method seems to be defined in each test. Lets move the common code to produce_consume_validate.py file.
One more observation: As per the current implementation, the method name says `sum_of_transactions` but its returning the addition of high-water-marks across the partitions. Lets make sure to reflect the sum of transactions across the partitions.